scripts/test/run-tests.sh

```diff
@@ -60,20 +55,32 @@ if [ "$1" == "--prebuilt-image" ]; then
 
 else
   if [ "$1" == "--all" -o "$1" == "--only" -a "$2" == "qemux86_64" ]; then
+    wget --progress=dot:giga -N ${UBUNTU_IMAGE_URL} -P input/
     convert_and_test "qemux86_64" \
                      "release-1" \
-                     "${UBUNTU_IMAGE_URL}" \
-                     "${UBUNTU_IMAGE}" \
-                     "${UBUNTU_IMAGE}.gz" \
-                     "configs/qemux86-64_config" || test_result=$?
+                     "input/Ubuntu-Bionic-x86-64.img.gz" \
+                     "gz" \
+                     "--config configs/qemux86-64_config" || test_result=$?
+
+    convert_and_test "qemux86_64" \
+                     "release-1" \
+                     "input/Ubuntu-Bionic-x86-64.img" \
+                     "gz" \
+                     "--config configs/qemux86-64_config" || test_result=$?
   fi
 
   if [ "$1" == "--all" -o "$1" == "--only" -a "$2" == "raspberrypi3" ]; then
+    wget --progress=dot:giga -N ${RASPBIAN_IMAGE_URL} -P input/
     convert_and_test "raspberrypi3" \
                      "release-1" \
-                     "${RASPBIAN_IMAGE_URL}" \
-                     "${RASPBIAN_IMAGE}.img" \
-                     "${RASPBIAN_IMAGE}.zip" \
+                     "input/2019-09-26-raspbian-buster-lite.zip" \
+                     "zip" \
+                     "--config configs/raspberrypi3_config" || test_result=$?
+
+    convert_and_test "raspberrypi3" \
+                     "release-1" \
+                     "input/raspbian_lite-2019-09-30/2019-09-26-raspbian-buster-lite.img" \
```

Comment:
Hm, I have not noticed this before but it seems that we are modifying the "input" files, e.g `gunzip` will remove the compressed file and replace it with the decompressed. I would object to this and leave the input file intact, and put the decompressed file in `work` directory.

Reply:
Yes we are.
I agree. That is much cleaner.
Fixing.